### DISCUSSION TITLE
Could Hexo's installation commands be simpler?

### DISCUSSION THREAD

**Discussion by bl-sun** [hexojs/hexo#5489] [Ideas]
I found that installing Hexo requires only the following 3 commands.

```shell
npx hexo init blog
cd blog
npx hexo server
```

But the homepage prompts us to use the following 5 commands.

```shell
npm install hexo-cli -g
hexo init blog
cd blog
npm install
hexo server
```

And the `npm install hexo` command in the figure below doesn't seem to be necessary.

![image](https://github.com/hexojs/hexo/assets/57904781/fec65abe-f49d-4e1f-b0fb-ba2eb18d8c99)

**Comment by stevenjoezhang** [comment]
You cannot run `npx hexo init blog` on a brand new machine. If you have not installed Hexo previously, `npx hexo` will report a command not found error.

**Reply by bl-sun** [reply]
![image](https://github.com/hexojs/hexo/assets/57904781/e0d2e399-aab8-4b8d-a022-3b2b91b23df7)
I learned about the npx command out of personal interest. I read from [his blog](https://www.ruanyifeng.com/blog/2019/02/npx.html) that the npx command temporarily installs the specified package. So I don't think you need the `npm install` command.

And then in the [official document](https://docs.npmjs.com/cli/v7/commands/npx) is described like this.
![image](https://github.com/hexojs/hexo/assets/57904781/47375fdb-fc27-4854-88fa-11cb4ceb6330)

I have installed a different version of nodejs using 'NVM for windows' to try it out myself with no error. I'm not sure I've ever installed hexo globally, though.

ps: 有道翻译的，不知道有没有问题。我看他们的描述，应该是可以的。我用nvm for windows安装了其他版本的nodejs，直接使用npx hexo是可以的。但是我不太确定，我在之前版本上全局安装hexo是不是有影响。

**Reply by stevenjoezhang** [reply]
很有意思，我第一次知道npx还有这个特性。不过我仍然建议通过`npm i -g hexo`进行安装，因为这样可以直接执行hexo命令（更加简单）。如果通过npx hexo 安装，每次执行都需要加npx 前缀。

I still recommend installing via `npm i -g hexo` because it allows you to execute the hexo command directly (which is simpler). If you install using `npx hexo`, you'll need to prefix the command with `npx` each time you run it.

**Comment by stevenjoezhang** [comment]
The documentation "Advanced installation and usage" is a little confusing. The prerequisite for `npm install hexo` is that you create a workspace in a new directory and have not previously installed hexo globally or in that workspace. If the workspace was created using `hexo init`, then `npm install hexo` is no longer necessary.

**Comment by uiolee** [comment]
the docs is a little obsolete. Maybe it need being updated
